web/pgadmin/browser/server_groups/servers/databases/static/js/database.ui.js

```diff
@@ -148,13 +150,21 @@ export default class DatabaseSchema extends BaseUISchema {
         editable: false, type: 'select', group: gettext('Definition'),
         readonly: function(state) {return !obj.isNew(state); },
         controlProps: { allowClear: false },
-        options: [{
-          label: gettext('icu'),
-          value: 'icu',
-        }, {
-          label: gettext('libc'),
-          value: 'libc',
-        }],
+        options: function() {
+          let options = [{
+            label: gettext('icu'),
+            value: 'icu',
+          }, {
+            label: gettext('libc'),
+            value: 'libc',
+          }];
+          if(obj.getServerVersion() >= 170000) {
+            options.push({
+              label: gettext('builtin'), value: 'builtin',
+            });
+          }
+          return Promise.resolve(options);
```

Comment:
Why do we need this? In the previous logic it was not there.

Reply:
We need this because the builtin locale provider is available on PG 17, so if the server version is below 17, the builtin option should not be available.